src/lakefs_spec/spec.py

```diff
@@ -758,6 +758,38 @@ def rm_file(self, path: str | os.PathLike[str]) -> None:
             # Directory listing cache for the containing folder must be invalidated
             self.dircache.pop(self._parent(path), None)
 
+    def touch(self, path: str | os.PathLike[str], truncate: bool = True, **kwargs: Any) -> None:
+        """
+        Create an empty file or update an existing file on a lakeFS server.
+
+        Parameters
+        ----------
+        path: str | os.PathLike[str]
+            The file path to create or update. Must be a fully qualified lakeFS URI.
+        truncate: bool
+            Whether to set the file size to 0 (zero) bytes, even if the path already exists.
+        **kwargs: Any
+            Additional keyword arguments to pass to ``LakeFSFile.open()``.
+
+        Raises
+        ------
+        NotImplementedError
+            If the targeted lakeFS server version does not support `touch()` operations.
+        """
+        with self.wrapped_api_call():
+            version_string = self.client.config_api.get_config().version_config.version
+            version_tuple = tuple(int(t) for t in version_string.split("."))
```

Comment:
Should we make this part of the constructor maybe and cache it? Having to guard code based on the server version might come up again in the future.

Reply:
Done in [71395c6](https://github.com/aai-institute/lakefs-spec/pull/214/commits/71395c6ba010c67da5a4021b252c4a97281e7b4e).